### PR TITLE
refactor: buildWhereClause with table.concat

### PR DESCRIPTION
**Pull request by artur-michalak**
## Description

Suggestion of buildWhereClause implementation.
Decide which version is easier to read.

## Checklist

- [X] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -80,10 +68,23 @@ local function buildWhereClause(filters)
                 placeholders[#placeholders+1] = filters.states[i]
                 statePlaceholders[i] = 'state = ?'
             end
-            query = query .. string.format(' AND (%s)', table.concat(statePlaceholders, ' OR '))
```
Comment (Manason) [inline server/main.lua:83]: Don't we still need the parentheses around the OR statement to isolate it from the AND?

**Review by Manason — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 1 above)
Comment (artur-michalak) [reply server/main.lua:83]: It would be good to verify this, but 90% no

Comment (artur-michalak) [reply server/main.lua:83]: As I tested, it worked well

Comment (Manason) [reply server/main.lua:83]: Logically `X AND Y OR Z` is not equal to `X AND (Y OR Z)` so how can this be the case?

Comment (artur-michalak) [reply server/main.lua:83]: In this case the statement starts with or
`Where status=1 or status=2 and x=3 and y=4`

Comment (Manason) [reply server/main.lua:83]: the AND would still take precedence https://mariadb.com/kb/en/operator-precedence/ so in the above example the query would return results where status=1 even if none of the other terms are true

Comment (artur-michalak) [reply server/main.lua:83]: thanks! Fix done

**Review by Manason — APPROVED** [review]
Nice job

**Pull request merged**